From f616a43ac74037e03e55fad9179d882d91bb38e4 Mon Sep 17 00:00:00 2001
From: Kunihito Higashiyama <kunihito.higashiyama.ur@renesas.com>
Date: Mon, 29 Jul 2013 12:28:30 +0900
Subject: [PATCH 0762/1083] ARM: shmobile: r8a7790: Fix MSIOF clock and device
 id

The duplication ID of MSIOF and QSPI is corrected.

Signed-off-by: Kunihito Higashiyama <kunihito.higashiyama.ur@renesas.com>
---
 arch/arm/mach-shmobile/clock-r8a7790.c |  2 +-
 arch/arm/mach-shmobile/setup-r8a7790.c | 10 +++++-----
 2 files changed, 6 insertions(+), 6 deletions(-)

diff --git a/arch/arm/mach-shmobile/clock-r8a7790.c b/arch/arm/mach-shmobile/clock-r8a7790.c
index 5a02bb2..778389c 100644
--- a/arch/arm/mach-shmobile/clock-r8a7790.c
+++ b/arch/arm/mach-shmobile/clock-r8a7790.c
@@ -349,7 +349,7 @@ static struct clk_lookup lookups[] = {
 	CLKDEV_CON_ID("usb_fck", &mstp_clks[MSTP703]),
 	CLKDEV_CON_ID("ss_usb", &mstp_clks[MSTP328]),
 	CLKDEV_DEV_ID("qspi.0", &mstp_clks[MSTP917]),
-	CLKDEV_DEV_ID("spi_sh_msiof.1", &mstp_clks[MSTP208]),
+	CLKDEV_DEV_ID("spi_sh_msiof.2", &mstp_clks[MSTP208]),
 	CLKDEV_DEV_ID("i2c-rcar.0", &mstp_clks[MSTP931]),
 	CLKDEV_DEV_ID("i2c-rcar.1", &mstp_clks[MSTP930]),
 	CLKDEV_DEV_ID("i2c-rcar.2", &mstp_clks[MSTP929]),
diff --git a/arch/arm/mach-shmobile/setup-r8a7790.c b/arch/arm/mach-shmobile/setup-r8a7790.c
index e2dbeea..3eb9881 100644
--- a/arch/arm/mach-shmobile/setup-r8a7790.c
+++ b/arch/arm/mach-shmobile/setup-r8a7790.c
@@ -914,7 +914,7 @@ static struct resource sh_msiof3_resources[] = {
 
 static struct platform_device sh_msiof0_device = {
 	.name		= "spi_sh_msiof",
-	.id		= 0,
+	.id		= 1,
 	.dev		= {
 		.platform_data	= &sh_msiof_info,
 	},
@@ -924,7 +924,7 @@ static struct platform_device sh_msiof0_device = {
 
 static struct platform_device sh_msiof1_device = {
 	.name		= "spi_sh_msiof",
-	.id		= 1,
+	.id		= 2,
 	.dev		= {
 		.platform_data	= &sh_msiof_info,
 	},
@@ -934,7 +934,7 @@ static struct platform_device sh_msiof1_device = {
 
 static struct platform_device sh_msiof2_device = {
 	.name		= "spi_sh_msiof",
-	.id		= 2,
+	.id		= 3,
 	.dev		= {
 		.platform_data	= &sh_msiof_info,
 	},
@@ -944,7 +944,7 @@ static struct platform_device sh_msiof2_device = {
 
 static struct platform_device sh_msiof3_device = {
 	.name		= "spi_sh_msiof",
-	.id		= 3,
+	.id		= 4,
 	.dev		= {
 		.platform_data	= &sh_msiof_info,
 	},
@@ -958,7 +958,7 @@ static struct spi_board_info spi_bus[] __initdata = {
 		.modalias	= "spidev",
 		.max_speed_hz	= 6000000,
 		.mode		= SPI_MODE_3,
-		.bus_num	= 1,
+		.bus_num	= 2,
 		.chip_select	= 0,
 	},
 };
-- 
1.8.3.2

